plugins/modules/vmware_guest.py

```diff
@@ -2207,7 +2207,7 @@ def customize_advanced_settings(self, vm_obj, config_spec):
                 self.change_detected = True
 
     def customize_customvalues(self, vm_obj):
-        if not self.params['customvalues']:
+        if not self.params['customvalues'] or not self.content.customFieldsManager:
```

Comment:
Actually, we can check if we connect to a vCenter or not. So my suggestion is:
```suggestion
        if not self.params['customvalues']:
            return

        if not self.is_vcenter():
            self.module.warn("It looks like you are connecting directly to an ESXi host. Since custom values are a vCenter feature, the parameter will be ignored.")
```

Reply:
Make sense.